table/pit-cs-tree.go

```diff
@@ -357,11 +368,22 @@ func (p *pitCsTreeNode) findMatchingDataCSPrefix(interest *ndn.Interest) CsEntry
 		return p.csEntry
 	}
 
-	if p.depth < interest.Name().Size() {
-		for _, child := range p.children {
-			if interest.Name().At(p.depth).Equals(child.component) {
-				return child.findMatchingDataCSPrefix(interest)
-			}
+	// if p.depth < interest.Name().Size() {
```

Comment:
Commented out code is the original code. I believe it doesn't work properly, as I wrote a test case based on your comment from the earlier PR: https://github.com/danningyu/YaNFD/pull/2#discussion_r815635384, and this code fails the test case. I wrote new code below that's correct, implementing all subname matching (ASNM).

ASNM definition from https://named-data.net/wp-content/uploads/2019/02/19-IEEE-NDNTables.pdf: "querying any Data whose name starts with this Interest name, is implemented in this scenario."

Reply:
Then, how do you test the equality on name components?
Seems that your code will return /a/c/v=10 for interest /a/b

Reply:
Hm good point, I will modify it. But am I correct that the original code doesn't work properly? It doesn't seem to work with the example you gave.

Reply:
I don't know about the original one. It looks good.

Reply:
The original is here, which was copied into the refactored code: https://github.com/named-data/YaNFD/blob/master/table/pit-cs.go#L279

Walking through the case where there is data in the CS for `/a/b/v=10` and an interest for `/a/b` arrives.

First, it calls f`indExactMatchEntry`, which will return the node for `b`, since it's an exact match with the `/a/b` prefix. Then, it'll call `findMatchingDataCSPrefix` with this `b` node, and the check `if p.depth < interest.Name().Size()` will not return true since `p.depth = 2` but `interest.Name.Size() = 2` as well. As a result, we'll never get to the `v=10` PitCS node that has the data for `/a/b/v=10`. 

Is this a correct understanding of the code?

Reply:
I think you are right. When `p.depth` is greater than `interest.Name.Size()`, it should return any of its children.

Reply:
I added an explicit check for if `p.depth >= interest.Name().Size()`, could you take a look again at the changes? Thanks!